File: Back.QuizApp/Controllers/Mappers/QuizMapper.cs
using Ynov.QuizApp.Business.Models;
using Ynov.QuizApp.Controllers.DTOs;

namespace Ynov.QuizApp.Controllers.Mappers;

public class QuizMapper {
    public QuizDTO ToDto(Quiz quiz) {
        return new QuizDTO() {
            Id = quiz.Id,
            Description = quiz.Description,
            Difficulty = quiz.Difficulty,
            Name = quiz.Name,
            CategoryId = quiz.Category.Id,
            CreationDate = quiz.CreationDate
        }; 
    }
}

File: Front.QuizApp/src/app/business/mappers/quiz.mapper.ts
import {Injectable} from '@angular/core';
import {QuizDto} from '../dtos/quiz.dto';
import {Quiz} from '../models/quiz.model';

@Injectable({providedIn: 'root'})
export class QuizMapper {
    public mapQuizFromApiToModel(apiQuiz: QuizDto): Quiz {
        return {
            id: apiQuiz.id,
            name: apiQuiz.name,
            description: apiQuiz.description,
            difficulty: apiQuiz.difficulty,
            creationDate: new Date(apiQuiz.creationDate),
            categoryId: apiQuiz.categoryId,
        };
    }
}
